### DISCUSSION TITLE
EPSG Database Pings and Updates

### DISCUSSION THREAD

**Discussion by nicklamkey** [pyproj4/pyproj#1346] [General]
I have a custom CRS that was just recently assigned an EPSG code and is now visible in the EPSG database. Does pyproj.CRS method ping the database or is there a file that is updated semi regularly that it pulls the data? As of now pyproj can't find my EPSG code so just trying to get an idea of how the EPSG codes work in pyproj and how often they are updated.

**Comment by djhoese** [comment]
I believe pyproj should always be asking for that type of information through the PROJ library. The EPSG database is usually updated in each minor(?) PROJ release so whether or not your EPSG code is recognized is dependent on its support in upstream PROJ. There is a chance I'm wrong, but from what I understand of your question I think this is correct.

**Reply by nicklamkey** [reply]
OK I think that makes sense to me. So PROJ doesn't retrieve CRS information from the EPSG data base on the fly as in it's not getting the information through EPSG API live. There's a separate file within the PROJ installation that contains all the CRS information keyed on EPSG code? I'm just trying to figure out how to allow people to programmatically access my CRS through python now that it has an EPSG code.

**Reply by djhoese** [reply]
Correct. As far as I know PROJ builds its local database from whatever EPSG (and other authorities) sources it has. The latest pyproj release used PROJ 9.3.0.

Just curious, do your users/people *need* to use EPSG codes? It might be the simplest way to identify something in the EPSG registry, but there are other ways to describe a CRS so I'm mostly just curious.

**Reply by nicklamkey** [reply]
No they don't need EPSG codes but it was heavily requested. Prior to getting on the EPSG registry we just shared the .prj file for ESRI and the WKT text for opensource. But evidently people relate to the EPSG codes more and link that with programmatic access to the CRS so that's what I'm attempting to do.

**Reply by nlamkey** [reply]
It does look like my CRS Is in the PROJ SQL database on Github. When do you think the next update of pyproj database will be?

**Reply by djhoese** [reply]
Do you know what version those PROJ updates will be released as? You should be able to install PROJ on your system or whatever prefix your Python is installed in and then install pyproj with `pip install pyproj --no-binary pyproj`. You may need to point PROJ_DIR to the location of your PROJ installation. See https://pyproj4.github.io/pyproj/stable/installation.html#from-pypi for more information.

The point is that pyproj wheels are bundled with a version of pyproj, but it can also be told to build against another PROJ installation. By telling pip to install pyproj from source, it should try to search for this other PROJ automatically. Unless there were major changes in the PROJ version that includes the database updates you need, things should "just work".

And to be clear, pyproj doesn't have its own database. It accesses the PROJ database through PROJ's APIs. This fact gets confusing sometimes when the bundled PROJ is involved.

**Reply by nlamkey** [reply]
I believe the version of PROJ that has my CRS is 10.096. Thanks for your help and the information.
